src/main/java/ArrayIndexedBag.java

```diff
@@ -0,0 +1,176 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ArrayIndexedBag<T> implements IndexedBag<T> {
+
+    private static final int DEFAULT_CAPACITY = 100;
+    private static final int NOT_FOUND = -1;
+    protected T[] bag;
+    protected int rear;
+
+    public ArrayIndexedBag() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    @SuppressWarnings("unchecked")
+    public ArrayIndexedBag(int initialCapacity) {
+        bag = (T[]) new Object[initialCapacity];
+        rear = 0;
+    }
+
+    /**
+     * Doubles the size of the bag array and copy the
+     * contents over.
+     */
+    @SuppressWarnings("unchecked")
+    private void expandCapacity() {
+        T[] newBag = (T[]) new Object[bag.length * 2];
+        for (int i = 0; i < bag.length; ++i) {
+            newBag[i] = bag[i];
+        }
+        bag = newBag;
+    }
+
+    /**
+     * Shifts elements in an array down (towards index 0) towards
+     * the starting index specified. This method assumes that
+     * the calling function manages the rear field.
+     *
+     * @param start Index of where shift starts/stops and element
+     *              overwritten.
+     */
+    private void shiftLeft(int start) {
+        for (int i = start; i < bag.length - 1; ++i) {
+            bag[i] = bag[i + 1];
+        }
```

Comment:
Should only need to go to `i < rear`, though logically identical. You'd just be wasting time assigning `null` to `null` on the unused portion of the array.